subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -230,6 +230,46 @@ int nrf_cloud_encode_ua(const struct nrf_cloud_ua_param *input,
 	return 0;
 }
 
+int nrf_cloud_encode_shadow_data(const struct nrf_cloud_sensor_data *sensor,
+				 struct nrf_cloud_data *output)
+{
+	int ret;
+	char *buffer;
+
+	__ASSERT_NO_MSG(sensor != NULL);
+	__ASSERT_NO_MSG(sensor->data.ptr != NULL);
+	__ASSERT_NO_MSG(sensor->data.len != 0);
+	__ASSERT_NO_MSG(output != NULL);
+
+	cJSON *root_obj = cJSON_CreateObject();
+	cJSON *state_obj = cJSON_CreateObject();
+	cJSON *reported_obj = cJSON_CreateObject();
+
+	if (root_obj == NULL) {
```

Comment:
Shouldn't we verify `state_obj` and `reported_obj` as well? They're even more likely to fail, as they are allocated after `root_obj`. And in case any of these allocations fail, we should free any memory that could've been already allocated before returning, see how `nrf_cloud_encode_ua` handles it.

Reply:
Fixed.